### PR TITLE
Disallow any dag tags longer than 100 char

### LINKED ISSUE #24820: Dag disappears when DAG tag is longer than 100 char limit
### Apache Airflow version

2.2.5

### What happened

We added new DAG tags to a couple of our DAGs. In the case when the tag was longer than the 100 character limit the DAG was not showing in the UI and wasn't scheduled. It was however possible to reach it by typing in the URL to the DAG.
Usually when DAGs are broken there will be an error message in the UI, but this problem did not render any error message.

This problem occurred to one of our templated DAGs. Only one DAG broke and it was the one with a DAG tag which was too long. When we fixed the length, the DAG was scheduled and was visible in the UI again.

### What you think should happen instead

Exclude the dag if it is over the 100 character limit or show an error message in the UI.

### How to reproduce

Add a DAG tag which is longer than 100 characters.

### Operating System

Ubuntu

### Versions of Apache Airflow Providers

_No response_

### Deployment

Other 3rd-party Helm chart

### Deployment details

Running Airflow in Kubernetes.
Syncing DAGs from S3 with https://tech.scribd.com/blog/2020/breaking-up-the-dag-repo.html

### Anything else

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by TommyDew42**
---
closes #24820.

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/models/dag.py [hunk 1]
```diff
@@ -347,6 +347,9 @@ def __init__(
     ):
         from airflow.utils.task_group import TaskGroup
 
+        if tags and any(len(tag) > 100 for tag in tags):
```
Comment (potiuk) [inline airflow/models/dag.py]: I just wonder - why 100 ? Do we know why the 100 causes the DAGs to disappear (in the original issue? ) I think having such a magic constant here is not good - it needs to come from somewhere ?

**Comment by eladkal** [comment]
I don't think this is right (or at least not valid to all backends)
I was able to create a DAG with tag longer than 100 chars with Sqlite (didn't test the others)
Screen Shot 2022-07-21 at 20 56 44

**Review by potiuk — CHANGES_REQUESTED** [review]
Yep. We need to confirm where the 100 limit comes from (and whether it's real) before we merge this one.

**Comment by anselboero** [comment]
I'd like to try to give an explanation :) 

The [DagTag](https://github.com/apache/airflow/blob/be7cb1e837b875f44fcf7903329755245dd02dc3/airflow/models/dag.py#L2781) class requires the maximum length of the Tag to be 100 chars.
The error occurs during the loading of the Tag into the Db.

I believe SqlLite is not affected by this bug as, according to the [SqlLite documentation](https://www.sqlite.org/datatype3.html) (3.1.1), 

> numeric arguments in parentheses that following the type name (ex: "VARCHAR(255)") are ignored by SQLite - SQLite does not impose any length restrictions [...] on the length of strings

It follows that the tag is uploaded into the db despite the restriction.

**Comment by potiuk** [comment]
> I'd like to try to give an explanation :)
> 
> The [DagTag](https://github.com/apache/airflow/blob/be7cb1e837b875f44fcf7903329755245dd02dc3/airflow/models/dag.py#L2781) class requires the maximum length of the Tag to be 100 chars. Because of that, the error occurs during the loading of the Tag into the Db.
> 
> I believe SqlLite is not affected by this bug as, according to the [SqlLite documentation](https://www.sqlite.org/datatype3.html) (3.1.1),
> 
> > numeric arguments in parentheses that following the type name (ex: "VARCHAR(255)") are ignored by SQLite - SQLite does not impose any length restrictions [...] on the length of strings
> 
> It follows that the tag is uploaded into the db despite the restriction.

Cool. Lets then extrat the 100 from there to a constant and use it in Both places.

**Comment by TommyDew42** [comment]
Thanks everyone's help! Updated made!

**Comment by TommyDew42** [comment]
I am a bit confused about the test [failure](https://github.com/apache/airflow/runs/7454890758?check_suite_focus=true) tho. Is the test flaky or we did break sth?

**Comment by potiuk** [comment]
You need to rebase. It was a flaky test that I fixed in #25231

**Comment by TommyDew42** [comment]
@potiuk Thanks. I actually never try `rebase` but always use `merge` LOL. Hope what I did fixed the tests.

**Comment by TommyDew42** [comment]
Oh wait. So I think the test was fixed when I merged the master into my branch. 😂 no need to do rebase.

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Well. you now have 58 commits in your change.. Rebase is REALLY better :)

**Comment by potiuk** [comment]
Highly recommend learning to rebase: https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#id14

File: airflow/models/dag.py [hunk 2]
```diff
@@ -374,6 +375,9 @@ def __init__(
     ):
         from airflow.utils.task_group import TaskGroup
 
+        if tags and any(len(tag) > TAG_MAX_LEN for tag in tags):
+            raise AirflowException("tag cannot be longer than 100 characters")
```
Comment (anselboero) [inline airflow/models/dag.py]: Sorry, short question,
should we map the Exception text to the TAG_MAX_LEN variable, too?

Comment (potiuk) [reply airflow/models/dag.py]: pls

**Comment by potiuk** [comment]
Merging. Random issue with MsSQL

**Pull request merged**